stacker/config/__init__.py

```diff
@@ -37,3 +37,164 @@ def parse_config(raw_config, environment=None):
     buff.seek(0)
     config = yaml.load(buff)
     return config
+
+
+class Dumper(yaml.Dumper):
+    pass
+
+
+class Loader(yaml.Loader):
+    pass
+
+
+def parse(raw, validate=True):
+    """Parse and validate a stacker config.
+
+    Args:
+        raw (str): the raw stacker configuration string in yaml format.
+        validate (bool): whether or not to validate the parsed config.
+
+    Returns:
+        :class:`Config`: the parse (and validated) stacker config.
+
+    """
+
+    config = yaml.load(raw, Loader=Loader) or Config()
+
+    if validate:
+        config.validate()
```

Comment:
I think it might be better if this returned a list of errors, rather than raising an exception, so we can print all of the errors at once.